npm/build-wiki.js

```diff
@@ -1,25 +1,44 @@
 #!/usr/bin/env node
+/* eslint-env node, es6 */
 // ---------------------------------------------------------------------------------------------------------------------
 // This script is intended to generate a wiki for this module
 // ---------------------------------------------------------------------------------------------------------------------
 
 require('shelljs/global');
-require('colors');
 
-// stop on encountering the first error
-set('-e');
-echo('-e', 'Generating wiki...');
-echo('-e', 'jsdoc2md');
+var fs = require('fs'),
 
-// some variables
-var OUT_DIR = 'out/wiki',
+    colors = require('colors/safe'),
+    jsdoc2md = require('jsdoc-to-markdown'),
+
+    OUT_DIR = 'out/wiki',
+    INP_DIR = 'lib/**/*.js',
     OUT_FILE = 'REFERENCE.md',
-    OUT_PATH = OUT_DIR + '/' + OUT_FILE;
+    OUT_PATH = OUT_DIR + '/' + OUT_FILE,
+    SUCCESS_MESSAGE = colors.green.bold(`- wiki generated at "${OUT_PATH}"`);
+
+module.exports = function (exit) {
+    console.log(colors.yellow.bold('Generating wiki using jsdoc2md...'));
+
+    try {
```

Comment:
There's an async function inside the try catch. This is an anti-pattern.

@kunagpal, what do you think is the best way of fixing it?